udemy_exercises/songs/Album.java

```diff
@@ -1,5 +1,71 @@
-package udemy_exercises.songs;
 
-class Album {
+import java.util.ArrayList;
+import java.util.LinkedList;
+
+public class Album {
+    private String name;
+    private String artist;
+    private ArrayList<Song> songs;
+
+    public Album(String name, String artist) {
+        this.name = name;
+        this.artist = artist;
+        this.songs = new ArrayList<Song>();
+    }
+
+    public boolean addSong(String title, double duration) {
+        if(findSong(title) == null) {
+            this.songs.add(new Song(title, duration));
+            return true;
+        }
+        return false;
+    }
+
+    private Song findSong(String title) {
+        for(Song checkedSong: this.songs) {
+            if(checkedSong.getTitle().equals(title)) {
```

Comment:
Currently, 'Title' is what makes each Song unique.  What if we wanted to support having both a standard version and an extended version of the same song on the Album.  In that case, uniqueness would be based on both the Title and the Duration.

In your currently implementation, this requires adding another condition to the 'if' statement.  If something then requires an additional condition, this code will become difficult to read.

In one of your Classes, you implemented an override of toString().  Is there a similar approach you could implement here to leverage core Java functionality, encapsulating the conditions in this line of code (and elsewhere that requires this same test)?